Doc/whatsnew/3.10.rst

```diff
@@ -611,6 +611,10 @@ Porting to Python 3.10
   :ref:`Python Path Configuration.  <init-path-config>`.
   (Contributed by Victor Stinner in :issue:`42260`.)
 
+* Cast the result of :c:func:`PyList_SET_ITEM`, :c:func:`PyTuple_SET_ITEM` and
```

Comment:
I would say that PyList_SET_ITEM etc are no longer rvalues and have type void. Or something like.

Casting to void is an implementation detail. What is important to the user that they can no longer write `x = PyList_SET_ITEM(a, b, c)` or `PyList_SET_ITEM(a, b, c) = x`.